### ISSUE TITLE
CoreCompileInputs doesn't include the name of the project in the name, possibly causing clashes

### ISSUE THREAD

**Issue by davkean** [Microsoft/msbuild#1645]
Most files that get generated into obj, include the project name to avoid clashes:

```
Debug\ConsoleApp146.AssemblyInfo.cs
Debug\ConsoleApp146.csproj.FileListAbsolute.txt
Debug\ConsoleApp146.csprojResolveAssemblyReference.cache
Debug\ConsoleApp146.dll
Debug\ConsoleApp146.pdb
ConsoleApp146.1.0.0.nuspec
ConsoleApp146.csproj.nuget.g.props
ConsoleApp146.csproj.nuget.g.targets
```

CoreCompileInputs.cache does not:

```
CoreCompileInputs.cache
```

This is going to lead to clashes when projects share intermediates.

**Comment by davkean** [comment]
Make note this affects projects that live in the same directory: https://github.com/dotnet/roslyn-project-system/issues/1528.

**Comment by Nazgul07** [comment]
I can confirm this issue... I run into build issues because of this. We have a solution with 120+ projects in it and the CoreCompileInputs.cache file not being uniquely named is causing issues.

**Comment by rainersigwald** [comment]
Fixed by #1783.
